rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/EditDynamicParametersWindow.java

```diff
@@ -102,34 +102,36 @@ protected void handleAdditionalInfrastructureFormItems(Map<String, String> selec
-        List<PluginDescriptor.Field> allDynamicFields = new LinkedList<>();
-        allDynamicFields.addAll(policyDynamicFields);
-        allDynamicFields.addAll(infrastructureDynamicIFields);
-
+        List<String> policyDynamicFieldFullNames = focusedPolicyPlugin.getConfigurableFields()
+                                                                      .stream()
+                                                                      .filter(PluginDescriptor.Field::isDynamic)
+                                                                      .map(field -> focusedPolicyPlugin.getPluginName() +
+                                                                                    field.getName())
+                                                                      .collect(Collectors.toList());
+        List<String> infrastructureDynamicFieldFullNames = focusedInfrastructurePlugin.getConfigurableFields()
+                                                                                      .stream()
+                                                                                      .filter(PluginDescriptor.Field::isDynamic)
+                                                                                      .map(field -> focusedPolicyPlugin.getPluginName() +
+                                                                                                    field.getName())
+                                                                                      .collect(Collectors.toList());
+        List<String> allDynamicFieldFullNames = new LinkedList<>();
+        allDynamicFieldFullNames.addAll(policyDynamicFieldFullNames);
+        allDynamicFieldFullNames.addAll(infrastructureDynamicFieldFullNames);
+
+        // this list will have all the initial items plus one hidden item for
+        // each item that will be disabled by the following code. The item
+        // ordering is preserved while inserting hidden items.
         List<FormItem> allFormItemsWithHiddenFields = new LinkedList<>();
 
         for (FormItem formItem : this.allFormItems) {
             allFormItemsWithHiddenFields.add(formItem);
-            if (allDynamicFields.stream()
-                                .noneMatch(field -> isPartOfDynamicFields(focusedInfrastructurePlugin,
-                                                                          focusedPolicyPlugin,
-                                                                          formItem,
-                                                                          field))) {
-
+            if (allDynamicFieldFullNames.stream()
+                                        .noneMatch(fieldFullName -> isItemEqualToDynamicField(formItem,
+                                                                                              fieldFullName))) {
                 if (this.allFormItemsPerPlugin.keySet()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1066)